pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -233,29 +250,45 @@ def __validate_init_configuration(self):
                     ' You can save the last checkpoint with ModelCheckpoint(save_top_k=None, monitor=None)'
                 )
 
-    def __init_ckpt_dir(self, filepath, save_top_k):
-        self._fs = get_filesystem(filepath if filepath is not None else "")
+    def __init_ckpt_dir(self, filepath, dirpath, filename, save_top_k):
+        if filepath:
+            if (dirpath or filename):
+                raise MisconfigurationException(
+                    f'filepath={filepath}, dirpath={dirpath} and'
+                    f' filename={filename} is not a valid configuration.'
```

Comment:
```suggestion
                    'You have set all three path/name inputs with are not feasable.'
                    f'You have to chose filepath="{filepath}" OR dirpath="{dirpath}: and filename="{filename}" configuration.'
```

Reply:
Should a deprecation message be included here as well?